tutorials/intermediate/legend_guide.py

```diff
@@ -241,7 +241,7 @@ class AnyObject(object):
     pass
 
 
-class AnyObjectHandler(object):
+class AnyObjectHandler(HandlerBase):
```

Comment:
Isn't the idea of this example to show that you can use any object (hence the name) and do not actually need to subclass any existing object?

Reply:
The "AnyObject" is referring to the idea that you can pass any object to the legend as a handle. The Legend handler should still be a child of HandlerBase.

Reply:
No, this example is meant to show that you may use any object as handler as long as it implements the `legend_artist` method. There also seems no reason to subclass `HandlerBase`, why "should" it? 

The next example then shows how to subclass `HandlerBase` for the ellipse case. This is as intended, because it actually uses some of the parent's methods.

Reply:
You should subclass HandlerBase because the legend expects to be dealing with HandlerBase's. It should be enough to modify legend and HandlerBase to change handler functionality. That's why I made AnyObjectHandler a HandlerBase in the first place. I made Legend call a new function. It doesn't make sense to have legend interact with any arbitrary random object you want because it makes it more difficult to update in the future. All you loose by sub-classing HandlerBase is the bad design of not sub-classing it.

Reply:
The text above that example reads:

>A custom handler can be implemented to turn any handle into a legend key (handles don’t necessarily need to be matplotlib artists). The handler must implement a “legend_artist” method which returns a single artist for the legend to use. Signature details about the “legend_artist” are documented at legend_artist().

Are you proposing to change this text as well? Otherwise this code example shows exactly what the text says; and it's good the way it is, because it allows people to easily grasp the concept. 

> I made Legend call a new function. 

Maybe you shouldn't do that.

Reply:
I agree with @ImportanceOfBeingErnest , this is showing off duck typing (we try to use `isinstance` as rarely as possible).

If this PR is expanding the required API for things that can be handlers, then that should be documented as such.